src/main/scala/com/cognite/spark/datasource/CdpRelation.scala

```diff
@@ -22,31 +33,66 @@ case class DeleteItem(
     id: Long
 )
 
-abstract class CdpRelation[T: Decoder](config: RelationConfig, shortName: String)
+case class PushdownFilter(fieldName: String, value: String)
+
+abstract class CdpRelation[T <: Product: Decoder](config: RelationConfig, shortName: String)
     extends BaseRelation
     with TableScan
     with Serializable
+    with PrunedFilteredScan
     with CdpConnector {
   @transient lazy private val itemsRead =
     MetricsSource.getOrCreateCounter(config.metricsPrefix, s"$shortName.read")
 
+  def fieldsWithPushdownFilter: Seq[String] = Seq[String]()
+
   val sqlContext: SQLContext
-  override def buildScan(): RDD[Row] =
+  override def buildScan(): RDD[Row] = buildScan(Array.empty, Array.empty)
+
+  override def buildScan(requiredColumns: Array[String], filters: Array[Filter]): RDD[Row] = {
+    val pushdownFilters: Seq[PushdownFilter] = fieldsWithPushdownFilter
+      .map(f => (f, filters.flatMap(getFilter(_, f)))) // map from fieldname to its filtering values
+      .flatMap(pdf => pdf._2.map(v => PushdownFilter(pdf._1, v)))
+      .distinct
+
     CdpRdd[T](
       sqlContext.sparkContext,
       (e: T) => {
         if (config.collectMetrics) {
           itemsRead.inc()
         }
-        toRow(e)
+        if (requiredColumns.isEmpty) toRow(e) else toRow(e, requiredColumns)
```

Comment:
I think we can put this logic into `toRow(e, requiredColumns)` and handle requiredColumns being empty in toRow.

Reply:
You're right, thanks.